Fix run soft-locking & reorganize bingus logic

Headline opened the pull request:
Howdy.

Found this mod from a friend who thought it was hilarious, on the first run we used the mod on we ended up in a situation where the only two monsters left in the game were bosses. This resulted in the run being soft-locked and we were unable to continue as the boss ended up being on our team.

With the old selection method, it would just retry another random mob if the originally selected mob was the boss. The following patch changes this strategy into one where we instead shuffle the list of mobs and loop until we find a suitable one. If we cannot, then we can just return out and prevent the mob from doing anything.

I'm sure you threw this together pretty quickly, but I took some time to organize it how I thought made sense, if this is unwelcome let me know and we can revert some of my organizational changes (or feel free to commit on this PR this yourself as you wish).

Thanks!

Headline committed:
Fix run soft-locking & reorganize bingus logic

Headline committed:
Version Bump

Headline committed:
Remove broken visual studio solution

bonixsteam-alt commented:
Thanks for the help, but it seems we both hit the same roadblock, because
it only detects if the monster is a boss, and it should detect if it is in
a BossGroup. Most likely it can't see if the entity is a boss because it
would mess with the item spawning logic (killing any monster in the Horde
Of Many would spawn an item). For some odd reason it seems to ignore that a
BossGroup is present, and because of that I've been stuck and couldn't fix
the softlock (and also I was a bit lazy).

Unfortunately I'm not home right now, but when I come home I will push the
changes to the beta branch.

And yes, I kind of wrote it quickly and also it's my first mod so that's
why I used a FixedUpdate instead of some other hook.

Again, thanks for the help.

śr., 2 cze 2021, 05:48 użytkownik Headline ***@***.***>
napisał:

> Howdy.
>
> Found this mod from a friend who thought it was hilarious, on the first
> run we used the mod on we ended up in a situation where the only two
> monsters left in the game were bosses. With the old selection method, it
> would just retry another random mob if the originally selected mob was the
> boss. The following patch changes this strategy into one where we instead
> shuffle the list of mobs and loop until we find a suitable one. If we
> cannot, then we can just return out and prevent the mob from doing anything.
>
> I'm sure you threw this together pretty quickly, but I took some time to
> organize it how I thought made sense, if this is unwelcome let me know and
> we can revert some of my organizational changes (or feel free to commit on
> this PR this yourself as you wish).
>
> Thanks!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/bonixsteam-alt/RoR2Bingus/pull/1
> Commit Summary
>
>    - Fix run soft-locking & reorganize bingus logic
>
> File Changes
>
>    - *M* BingusMod/Assets.cs
>    <https://github.com/bonixsteam-alt/RoR2Bingus/pull/1/files#diff-f2959af2f79402f443325e82dbbb7b2475c49326e8c5c911c00550e802fb4d2c>
>    (70)
>    - *M* BingusMod/BingusMod.cs
>    <https://github.com/bonixsteam-alt/RoR2Bingus/pull/1/files#diff-79a990518f389e12e2f23af204991ec28299908b6041e6466d8bd1db5325e389>
>    (103)
>
> Patch Links:
>
>    - https://github.com/bonixsteam-alt/RoR2Bingus/pull/1.patch
>    - https://github.com/bonixsteam-alt/RoR2Bingus/pull/1.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bonixsteam-alt/RoR2Bingus/pull/1>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARI5BCOX247QKGCIXA4V2PLTQWSXFANCNFSM45545PLQ>
> .
>

bonixsteam-alt committed:
Reorganized

Everything seems to work but after some time it stops working (left in chat loggers for 1, convenience and 2, well, testing)

bonixsteam-alt committed:
Adjusted display position

Everything seems to be fine, and I think what I said in the last commit was a one time thing, but if it persists, I will fix it.

bonixsteam-alt commented:
I think it's all fixed and you have my thanks for helping me out with this, and I saw what you said on discord, yes, I'm new to RoR2 modding. Again really thank you for the help.
